RadarSDK/Radar.m

```diff
@@ -406,6 +407,18 @@ + (void)sendLogConversionRequestWithName:(NSString * _Nonnull) name
     }];
 }
 
++ (void)logOpenedAppConversion {
+    // if opened_app has been logged within the last second, don't log it again
+    NSTimeInterval lastAppOpenTimeInterval = [[NSDate date] timeIntervalSinceDate:[RadarSettings lastAppOpenTime]];
+    if (lastAppOpenTimeInterval > 1) {
+        [RadarSettings updateLastAppOpenTime];
+        [self sendLogConversionRequestWithName:@"opened_app" metadata:nil completionHandler:^(RadarStatus status, RadarEvent * _Nullable event) {
+            NSString *message = [NSString stringWithFormat:@"Conversion name = %@: status = %@; event = %@", event.conversionName, [Radar stringForStatus:status], event];
+            [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo message:message];
+        }];
+    }
+}
+
```

Comment:
As noted during lunch, we'll need a separate `logOpenedAppConversionWithMetadata` that customers can manually insert into their `UNUserNotificationCenterDelegate`. This function should:
- have priority over the other `logOpenedAppConversion` calls
- maintain the deduplication logic


Currently I have:
```+ (void)logOpenedAppConversionWithMetadata:(NSDictionary *_Nullable)metadata {
    // log that we're in this
    [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo message:@"logOpenedAppConversionWithMetadata"];
    // if opened_app has been logged in the last 1000 milliseconds, don't log it again
    NSTimeInterval lastAppOpenTimeInterval =[[NSDate date] timeIntervalSinceDate:[RadarSettings lastAppOpenTime]];
    if (lastAppOpenTimeInterval * 1000 > 1000) {
        [RadarSettings updateLastAppOpenTime];
        [self sendLogConversionRequestWithName:@"opened_app" metadata:metadata completionHandler:^(RadarStatus status, RadarEvent * _Nullable event) {
            NSString *message = [NSString stringWithFormat:@"Conversion name = %@: status = %@; event = %@", event.conversionName, [Radar stringForStatus:status], event];
            [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo message:message];
        }];
    }
}

+ (void)logOpenedNotificationConversionWithMetadata:(NSDictionary *_Nullable)metadata {
    [self sendLogConversionRequestWithName:@"opened_notification" metadata:metadata completionHandler:^(RadarStatus status, RadarEvent * _Nullable event) {
        NSString *message = [NSString stringWithFormat:@"Conversion name = %@: status = %@; event = %@", event.conversionName, [Radar stringForStatus:status], event];
        [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo message:message];
    }];
}
```
which seems to be prioritizing the other `app_opened` over the metadata one

Reply:
I take this back — I no longer think we should be unifying `notification_opened` and `app_opened`. Sending a note in channel now